app/components/TopicItem.jsx

```diff
@@ -46,9 +46,7 @@ export default class TopicItem extends Component {
 }
 
 TopicItem.propTypes = {
-  text: PropTypes.string.isRequired,
-  id: PropTypes.string.isRequired,
-  index: PropTypes.number.isRequired,
+  topic: PropTypes.object.isRequired,
```

Comment:
any reason for changing this other than LoC? passing in `id` and `text` makes props comparisons simpler (and indicate what is actually used).